### ISSUE TITLE
Working with client certificate Lua and EnvoyFilter in Istio environment

### ISSUE THREAD

**Issue by liavyona** [envoyproxy/envoy#15260]
*Title*: *Working with Lua and EnvoyFilter in Istio environment*

*Description*:
I'm trying to add the client certificate URL encoded as an header to my service using `EnvoyFilter`.
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: test-filter
  namespace: istio-system
spec:
  configPatches:
  - applyTo: HTTP_FILTER
    match:
      context: GATEWAY
      listener:
        filterChain:
          filter:
            name: "envoy.http_connection_manager"
            subFilter:
              name: "envoy.router"
    patch:
      operation: INSERT_BEFORE
      value: # lua filter specification
        name: envoy.filters.http.lua
        typed_config:
          "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua
          inlineCode: |
            function envoy_on_request(handle)
              local headers = handle:headers()
              local streamInfo = handle:streamInfo()
              dsc = streamInfo:downstreamSslConnection()
              if dsc ~= nil then
                if dsc:peerCertificatePresented() then
                  headers:add("x-peer-given", "yes")
                else
                  headers:add("x-peer-given", "no")
                end
                headers:add("x-ssl-client-certificate", dsc:urlEncodedPemEncodedPeerCertificateChain())
              end
            end
```
And when I try to address to my service I get that a peer certificate wasn't presented. 
The client request works in nginx which I pass the certificate using the following configuration:
`proxy_set_header X-SSL-Client-Certificate $ssl_client_escaped_cert;`
Am I missing something with Envoy? Is there a better way?

[optional *Relevant Links*:]
I used the [Lua filter docs](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter#urlencodedpemencodedpeercertificatechain).
[Nginx SSL embedded variables](http://nginx.org/en/docs/http/ngx_http_ssl_module.html)
Istio gateway:
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: istio-gateway
  namespace: istio-system
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - example.com
    port:
      name: https
      number: 443
      protocol: HTTPS
    tls:
      credentialName: gateway-certificate
      mode: SIMPLE
```

**Comment by lambdai** [comment]
Can you confirm the client (which is nginx I assume?) is using mtls?

proxy_set_header seems set the client cert in HTTP request, but `dsc:urlEncodedPemEncodedPeerCertificateChain` returns you the info from the ssl session. This function doesn't read http header provided by nginx

**Comment by liavyona** [comment]
Thank you @lambdai 
The client is not a nginx but a Go library.
The server used nginx as its proxy previously and now we changed it to envoy by integrating with Istio.
The SSL session is the same as the client sends a request with its certificate. With nginx we used `$ssl_client_escaped_cert` embedded variable to pass the client certificate to the actual server (from the nginx) as an header.
We want to do the same with envoy.
